File: Python/Sample3.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# MCP23008
# This code is designed to work with the MCP23008_I2CR8G5LE_10A I2C relay controller available from ControlEverything.com.
# https://www.controleverything.com/content/Relay-Controller?sku=MCP23008_I2CR8G5LE_10A#tabs-0-product_tabset-2

import smbus

# Get I2C bus
bus = smbus.SMBus(1)

# MCP23008 address, 0x20(32)
# Select IODIR register, 0x00(00)
#		0xFF(255)	All pins are configured as input
bus.write_byte_data(0x20, 0x00, 0xFF)

# MCP23008 address, 0x20(32)
# Select GPPU register, 0x06(06)
#		0xFF(255)	Pull-up enabled on all pins
bus.write_byte_data(0x20, 0x06, 0xFF)

# Output to screen
print "Pull-up enabled on all pins"

File: Java/SAMPLE3.java
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// MCP23008
// This code is designed to work with the MCP23008_I2CR8G5LE_10A I2C relay controller available from ControlEverything.com.
// https://www.controleverything.com/content/Relay-Controller?sku=MCP23008_I2CR8G5LE_10A#tabs-0-product_tabset-2

import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class SAMPLE3
{
	public static void main(String args[]) throws Exception
	{
		// Create I2C bus
		I2CBus bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, MCP23008 I2C address is 0x20(32)
		I2CDevice device = bus.getDevice(0x20);
		
		// Configure all pins as input
		device.write(0x00, (byte)0xFF);
		
		// Pull-up all pins
		device.write(0x06, (byte)0xFF);
		
		// Output to screen
		System.out.println("Pull-up enabled on all pins");
	}
}
